Remove outdated or harmful rules (#37)
guardian/recommendations: client-side.md

## Dependencies

- If starting a new project, favour ES6 modules unless you can justify
  otherwise.

> - Do not commit your dependencies into Git. Instead, lock your dependencies (direct and transitive) using a lockfile.

Reviewer: In some circumstances committing your dependencies can be helpful (e.g. committing the Yarn binary using [`yarn policies`](https://classic.yarnpkg.com/en/docs/cli/policies))

Author: Is that useful as a generalised recommendation or is it fairly specific to certain use-cases?

Reviewer: Yeah, it's very specific. Maybe we could soften to:

```suggestion
- Prefer locking your dependencies (direct and transitive) using a lockfile rather than committing them into Git.
```

- Don’t fail silently.
- Report your JavaScript errors into a logging system (e.g. Sentry).
- Define web performance budgets and track them

> ### Build tools/task runners

Author: ```suggestion
### Build tools/task runners

- Have an automated process to produce your final assets (e.g. webpack, rollup, microbundle ).
```

targets, automated testing or RUM?*
- Define web performance budgets and track them

### Build tools/task runners

- Have an automated process to produce your final assets (e.g. using

>   Grunt, Gulp, [Plumber](https://github.com/plumberjs/plumber), etc).

Reviewer: Why remove this one? The example tooling is a bit dated, admittedly.

Author: 👍  Have updated with a more modern tool list.
